src/main/java/htsjdk/tribble/index/DynamicIndexCreator.java

```diff
@@ -56,13 +57,18 @@
     MathUtils.RunningStat stats = new MathUtils.RunningStat();
     long basesSeen = 0;
     Feature lastFeature = null;
-    File inputFile;
+    // TODO: actually this field is not needed
+    Path inputPath;
```

Comment:
In this case, because it is package protected, I do not think that is going to harm at all (unless someone is subclassing in the same package outside HTSJDK).

If we are breaking compatibiity anyway, as it is unused, why don't we just remove the field (no the constructor arg)? It is only used for get the index creators...